horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -856,4 +858,47 @@ public Object updateSchema(Integer id, String path, String schemaUri) {
       em.flush();
       return schemas;
    }
+
+   private void onCalculateDataSets(String param) {
+      log.info("calculate_datasets");
+      String[] parts = param.split(";", 3);
+      if (parts.length < 3) {
+         log.errorf("Received notification to calculate dataset %s but cannot extract run ID.", param);
+         return;
+      }
+      int runId;
+      try {
+         runId = Integer.parseInt(parts[0]);
+      } catch (NumberFormatException e) {
+         log.errorf("Received notification to calculate dataset for run %s but cannot parse as run ID.", parts[0]);
+         return;
+      }
+      String role = parts[2].trim();
+      if (role.isBlank()) {
+         log.errorf("Received notification which provides invalid role information [%s]", role);
+         return;
+      }
+
+      try (@SuppressWarnings("unused") CloseMe h1 = roleManager.withRoles(em, parts[2])){
```

Comment:
Please use the token (`parts[1]`), too, otherwise you might break some use-case.

Reply:
ok.